libraries/abstractions/transport/secure_sockets/transport_secure_sockets.h

```diff
@@ -0,0 +1,188 @@
+     */
+    bool disableSni;
+
+    /**
+     * @brief Set this to a non-zero value to use TLS max fragment length
+     * negotiation (TLS MFLN).
+     *
+     * @note The network stack may have a minimum value for this parameter and
+     * may return an error if this parameter is too small.
+     */
+    size_t maxFragmentLength;
+
+    const char * pRootCa;     /**< @brief String representing a trusted server Root CA certificate. */
+    size_t rootCaSize;        /**< @brief Size associated with #IotNetworkCredentials_t.pRootCa. */
+    const char * pClientCert; /**< @brief String representing the client certificate. */
+    size_t clientCertSize;    /**< @brief Size associated with #IotNetworkCredentials_t.pClientCert. */
+    const char * pPrivateKey; /**< @brief String representing the client certificate's private key. */
+    size_t privateKeySize;    /**< @brief Size associated with #IotNetworkCredentials_t.pPrivateKey. */
+} SocketsConfig_t;
+
+
+/**
+ * @brief Sets up a TLS session on top of a TCP connection using the Secure Sockets API.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ * @param[in] pServerInfo Server connection info.
+ * @param[in] pSocketsConfig socket configs for the connection.
+ *
+ * @return #TRANSPORT_SOCKET_STATUS_SUCCESS on success;
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, #TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY,
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, #TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR,
+ *         #TRANSPORT_SOCKET_STATUS_DNS_FAILURE, #TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE on failure.
+ */
+TransportSocketStatus_t SecureSocketsTransport_Connect( NetworkContext_t * pNetworkContext,
+                                                        const ServerInfo_t * pServerInfo,
+                                                        const SocketsConfig_t * pSocketsConfig );
+
+/**
+ * @brief Closes a TLS session on top of a TCP connection using the Secure Sockets API.
+ *
+ * @param[out] pNetworkContext The output parameter to end the TLS session and
+ *             clean the created network context.
+ *
+ * @return #TRANSPORT_SOCKET_STATUS_SUCCESS on success;
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, #TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR on failure.
+ */
+int32_t SecureSocketsTransport_Disconnect( const NetworkContext_t * pNetworkContext );
+
+
+/**
+ * @brief Receives data over an established TLS session using the Secure Sockets API.
+ *
+ * This can be used as #TransportInterface.recv function for receiving data
+ * from the network.
+ *
+ * @param[in] pNetworkContext The network context created using Socket_Connect API.
+ * @param[out] pBuffer Buffer to receive network data into.
+ * @param[in] bytesToRecv Number of bytes requested from the network.
+ *
+ * @return Number of bytes received if successful; negative value on error.
+ */
```

Comment:
This API should mention that a return value of `0` indicates a receive timeout. Also, a `0` input value of `bytesToRecv` should be treated as an invalid input parameter.

Reply:
fixed in 5fabf463be948c1821f30e186dcb9a13009ba915

referencing https://github.com/FreeRTOS/FreeRTOS/blob/lts-development/FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/include/tls_freertos.h#L176-L178